### ISSUE TITLE
Unable to set app icon

### ISSUE THREAD

**Issue by williambuchanan2** [dotnet/maui#7715]
### Description

I don't seem to be able to set an icon for the app. I have tried the various options suggested, and selecting the icon file in the project properties, but no matter what I do I end up with the default '.NET. icon.

Also, in the android settings there seems to be a separate dropdown to select the icon - this dropdown is always empty.

### Steps to Reproduce

Create a new Maui project.
Set the app icon in the MauiIcon setting in the project file.
Run app - notice it shows a different icon.

Also...
In project properties
Android -> Manifest -> Application Icon
Notice this list is empty

### Version with bug

6.0 Release Candidate 3

### Last version that worked well

Unknown/Other

### Affected platforms

Android

### Affected platform versions

API 31

### Did you find any workaround?

No

**Comment by RichardMarsh** [comment]
For what it's worth I have the same issue but on GA

**Comment by stephenhauck** [comment]
I am up to date on VS 2022 Preview for Windows on Windows 10
Paired to an up to date MacBook Pro Max

I just tried to set the MauiIcon in a simple project generated from the template using a JPG file because the documentation states ...

_A .NET MAUI app icon can use any of the standard platform image formats, including Scalable Vector Graphics (SVG) files._

It doesn't work ... 

somehow it still uses the default icon graphic even though it's not in the project any longer ...
![image](https://user-images.githubusercontent.com/12648793/171847592-990977f2-c18f-4c04-876c-b752dd23649f.png)


I tried to delete the Debug folder to see if that helped and even after a full rebuild something isn't quite right
[Rebuildoutput.txt](https://github.com/dotnet/maui/files/8832080/Rebuildoutput.txt)

**Comment by stephenhauck** [comment]
ok .. I renamed my icon to match the template .. appicon.jpg and it's working 
It must be something with the changing of the image property to "MauiIcon" not cascading through the various platforms ?

**Comment by AndreasReitberger** [comment]
I can confirm this behavior. Only overwriting the original files does work.

Also having troubles with the splash. Changes to the splash.svg file will be applied only after removing the app from the phone. 

This happens for me on iOS, haven't tried Android yet.

**Comment by stephenhauck** [comment]
I did also make sure that running VS 2022 Preview as admin did not make a difference in the misbehavior.

**Comment by williambuchanan2** [comment]
> ok .. I renamed my icon to match the template .. appicon.jpg and it's working It must be something with the changing of the image property to "MauiIcon" not cascading through the various platforms ?

I just tried that and it seems to work. The only difference is that mine had created an svg file by default - not a jpg (appicon.svg).

**Comment by williambuchanan2** [comment]
Something really strange is that now I managed to get my icon to show (having updated the packaged icon) I can't get it to show changes to the icon. I just drew a red line through it and no matter what I do it still shows the old version. It is like the icon is being cached somewhere but you can't see where.

**Comment by RichardMarsh** [comment]
> ok .. I renamed my icon to match the template .. appicon.jpg and it's working It must be something with the changing of the image property to "MauiIcon" not cascading through the various platforms ?

Does it work on Android? My main issue is that no matter what I do I can't get the icon to show for Android.

**Comment by williambuchanan2** [comment]
> Does it work on Android? My main issue is that no matter what I do I can't get the icon to show for Android.
I managed to get it to work on Android by overwriting the appinfo.svg. However, once it picked up that file I was unable to get it to show any subsequent changes I made.

**Comment by stephenhauck** [comment]
I am having similar problems an Android ... wish MAUI was more "ready" they announced GA on May 24th .... I REALLY want to use it for some re-writes...

**Comment by stephenhauck** [comment]
FWIW - The app icon is in the start menu after running it locally on Windows 10

**Comment by Quietscheente** [comment]
I have no chance to set this appicon on Android, even if I override the existing one and use the same Name (and delete the existing foreground icon).

My .csproj file:
`<MauiIcon Include="Resources\AppIcon\appicon.svg" Color="#512BD4" />`

It only shows the android default app icon
![appicon](https://user-images.githubusercontent.com/57129685/172150889-de0e6e15-4cc2-4db0-89be-892f71e4096c.svg)
.

**Comment by Quietscheente** [comment]
> On Android, when `ForegroundFile` is not set, the icon doesn't work. #7203 Also, calling it anything other than `appicon` doesn't work.

Got it working. Thanks.

**Comment by jfversluis** [comment]
Duplicate of #7203

**Comment by williambuchanan2** [comment]
> Duplicate of #7203

This is not a duplicate of #7203. The only thing both have in common is their relation to the app icon.
